Cargo.toml

```diff
@@ -33,21 +33,22 @@ ark-serialize = { version = "0.4.2", default-features = false }
 hex = "0.4.3"
 byteorder = "1.4.3"
 
-# ethereum compat
-ethers-core = { version = "2.0.7", default-features = false, optional = true }
 
 # error handling
 thiserror = "1.0.39"
 color-eyre = "0.6.2"
 
+# ethereum compat
+ruint = { version = "1.12.3", default-features = false ,optional = true, features=["alloc", "ark-ff-04"] }
+
 cfg-if = "1.0.0"
 
 [dev-dependencies]
 criterion = "0.5.1"
 hex-literal = "0.4.1"
 tokio = { version = "1.29.1", features = ["macros"] }
 serde_json = "1.0.94"
-ethers = "2.0.7"
+alloy = { version = "0.6.4", default-features = false, features = ["contract", "node-bindings"] }
```

Comment:
Can we bump to `0.9`? This might have some breaking changes so we could also do this in another PR.

Reply:
yeah, I'll do it in another PR